chalice/package.py

```diff
@@ -136,10 +151,15 @@ def _generate_lambdafunction(self, resource, template):
             }
             lambdafunction_definition['Properties'].update(
                 reserved_concurrency_config)
-        if resource.layers:
+
+        layers = resource.layers or []  # type: List[Any]
+        if self._chalice_layer:
+            layers.append({'Ref': self._chalice_layer})
```

Comment:
this needs to be inserted at the start, we want user explicit layers to have precedence.